azext_iot/tests/dps/core/test_dps_discovery_unit.py

```diff
@@ -57,3 +58,46 @@ def test_get_target_by_cstring(self, fixture_cmd, get_mgmt_client):
         assert target["entity"] == parsed_fake_login["HostName"]
         assert target["policy"] == parsed_fake_login["SharedAccessKeyName"]
         assert target["primarykey"] == parsed_fake_login["SharedAccessKey"]
+
+    def test_get_target_by_host_name(self, fixture_cmd, get_mgmt_client):
```

Comment:
prefer `hostname` since it's a single term.

Reply:
there are a lot of host_name already

Reply:
but will change it here cause it does look better for the func name (same in other places)